### ISSUE TITLE
Can't change the resource pool of modules while ng build.

### ISSUE THREAD

**Issue by DerekLin86** [angular/angular-cli#9612]
### Versions
```
# ./node_modules/\@angular/cli/bin/ng -v

    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
    
Angular CLI: 1.6.8
Node: 8.9.4
OS: linux x64
Angular: 5.2.4
... animations, common, compiler, compiler-cli, core
... platform-browser, platform-browser-dynamic, upgrade

@angular/cli: 1.6.8
@angular-devkit/build-optimizer: 0.0.42
@angular-devkit/core: 0.0.29
@angular-devkit/schematics: 0.0.52
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.8
@schematics/angular: 0.1.17
typescript: 2.6.2
webpack: 3.10.0
```
* .angular-cli.json
```
{
    "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
    "project": {
        "name": "webcore"
    },
    "apps": [
    {
        "root": ".",
        "outDir": "dist",
        "index": "index.html",
        "main": "main.ts",
        "tsconfig": "tsconfig.app.json",
        "environmentSource": "environments/environment.ts",
        "environments": {
            "dev": "environments/environment.ts",
            "prod": "environments/environment.prod.ts"
        }
    }]
}
```
* tsconfig.app.json
```
cat tsconfig.app.json 
{
    "compilerOptions": {
        "baseUrl": ".",
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "module": "commonjs",
        "moduleResolution": "node",
        "noImplicitAny": true,
        "sourceMap": true,
        "suppressImplicitAnyIndexErrors": true,
        "target": "es6",
        "paths": {
            "*": [
                "RESOURCE_POOL/*"
            ]
        }
    },
    "exclude": [
        "RESOURCE_POOL/*",
        "node_modules/*"
    ]
}
```
* The modules under the folder RESOURCE_POOL
```
ls RESOURCE_POOL/
core-js               rxjs               zone.js               @angular          
node_modules     webpack
```

### Repro steps
I want to change the reference path of the modules which relate to build dependency to RESOURCE_POOL.
1. Adjust the configurations as above.
2. Execute angular-cli locally.
```
# ./node_modules/\@angular/cli/bin/ng build
```

### Observed behavior

- Only rxjs refers to the correct path:
  - main.bundle.js

```
.....
var root_1 = __webpack_require__("../../../../../RESOURCE_POOL/rxjs/util/root.js");
var isArrayLike_1 = __webpack_require__("../../../../../RESOURCE_POOL/rxjs/util/isArrayLike.js");
var isPromise_1 = __webpack_require__("../../../../../RESOURCE_POOL/rxjs/util/isPromise.js");
var isObject_1 = __webpack_require__("../../../../../RESOURCE_POOL/rxjs/util/isObject.js");
var Observable_1 = __webpack_require__("../../../../../RESOURCE_POOL/rxjs/Observable.js");
var iterator_1 = __webpack_require__("../../../../../RESOURCE_POOL/rxjs/symbol/iterator.js");
.....
```
- other modules refer to ./node_module, but I don't expect this situation.
```
....
//# sourceMappingURL=root.js.map
/* WEBPACK VAR INJECTION */}.call(exports, __webpack_require__("../../../../webpack/buildin/global.js")))
.....
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = __webpack_require__("../../../core/esm2015/core.js");
const platform_browser_1 = __webpack_require__("../../../platform-browser/esm2015/platform-browser.js");
const static_1 = __webpack_require__("../../../upgrade/esm2015/static.js");
.....
```

### Desired behavior
I expect the result as follows:
```
....
//# sourceMappingURL=root.js.map
/* WEBPACK VAR INJECTION */}.call(exports, __webpack_require__("../../../../../RESOURCE_POOL/webpack/buildin/global.js")))
.....
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = __webpack_require__("../../../../../RESOURCE_POOL/@angular/core/esm2015/core.js");
const platform_browser_1 = __webpack_require__("../../../../../RESOURCE_POOL/@angular/platform-browser/esm2015/platform-browser.js");
const static_1 = __webpack_require__("../../../../../RESOURCE_POOL/@angular/upgrade/esm2015/static.js");
.....
```
Is it a bug? Or I configure with the wrong way?
Please help me to resolve this issue, thank you.

**Comment by filipesilva** [comment]
This seems like a bug but we'll need to look at a reproduction to find and fix the problem. Can you setup a minimal repro please? 

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

**Comment by DerekLin86** [comment]
Please refer to this repository https://github.com/DerekLin86/cli_build_issue/tree/master

**Comment by DerekLin86** [comment]
Is there any solution?

**Comment by clydin** [comment]
A more complete implementation of TypeScript path mapping will be available in 7.0 (prerelease 7.0.0-beta.4 is currently available for testing).  If this version does not solve the identified problem, please consider opening a new issue detailing the versions and environment of the affected project.
